$OpenBSD: patch-src_sip_msg_c,v 1.1.1.1 2021/04/28 19:12:30 sthen Exp $

This isn't really "old libpcap", it's a local OpenBSD change to make
pcap files MI. Upstream libpcap does this differently by using timevals
directly, and converting in the file load/save routines instead.

Fix by removing the compile "fix" and copy the struct members rather
than returning them directly. Fixes hard loop when viewing INVITEs.

Index: src/sip_msg.c
--- src/sip_msg.c.orig
+++ src/sip_msg.c
@@ -108,8 +108,10 @@ msg_get_time(sip_msg_t *msg) {
     struct timeval t = { };
     frame_t *frame;
 
-    if (msg && (frame = vector_first(msg->packet->frames)))
-        return frame->header->ts;
+    if (msg && (frame = vector_first(msg->packet->frames))) {
+        t.tv_sec = frame->header->ts.tv_sec;
+        t.tv_usec = frame->header->ts.tv_usec;
+    }
     return t;
 }
 
